http/logical.go

```diff
@@ -77,9 +77,28 @@ func buildLogicalRequest(core *vault.Core, w http.ResponseWriter, r *http.Reques
 
 	case "POST", "PUT":
 		op = logical.UpdateOperation
-		// Parse the request if we can
-		if op == logical.UpdateOperation {
-			err := parseRequest(r, w, &data)
+
+		// First attempt to get form parameters; if none exist, read the body and
+		// parse as JSON
+		err := r.ParseForm()
```

Comment:
This parsing will not have the `max_request_size` limiter imposed. Is that something you'd want to put at the start, applicable to both parsing paths?

Reply:
Yeah, it's true. I guess I should put it behind that, it just makes things a bit more complicated.

Reply:
Fixed